cmd/zoekt-sourcegraph-indexserver/queue.go

```diff
@@ -128,6 +136,96 @@ func (q *Queue) Iterate(f func(*IndexOptions)) {
 	}
 }
 
+// debugIteratedOrdered will call f on each queueItem (sorted by indexing priority)
+// known to the queue, including queueItems that have been popped from the queue).
+//
+// Note: The mutex is held during all calls to f. Callers must not modify the queueItems.
+func (q *Queue) debugIteratedOrdered(f func(*queueItem)) {
+	q.mu.Lock()
+	defer q.mu.Unlock()
+
+	queueItems := make([]*queueItem, 0, len(q.items))
+	for _, item := range q.items {
+		queueItems = append(queueItems, item)
+	}
+
+	sort.Slice(queueItems, func(i, j int) bool {
+		x, y := queueItems[i], queueItems[j]
+
+		if x.heapIdx < 0 && y.heapIdx >= 0 {
+			// if x is popped, but y isn't - then put x at the back
+			return false
+		}
+
+		if y.heapIdx < 0 && x.heapIdx >= 0 {
+			// if y is popped, but x isn't - then ensure x is at the front
+			return true
+		}
```

Comment:
minor suggestion for avoiding the need to comment code:
```suggestion
		// popped items go to the back
		xPopped := x.heapIdx < 0
		yPopped := y.heapIdx < 0
		if xPopped != yPopped {
			return !xPopped
		}
```

it might even read better as `xEnqueued` or `xOnQueue`, then the return is easier to understand.